From 2978e6986a0db1b83ebb8cfec43b83fda38faaac Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Tue, 20 May 2014 11:08:54 -0700
Subject: [PATCH 038/117] atomisp: Make firmware mismatches non-fatal

The atomisp driver is very strict about firmware versions, and would
throw a panicking assert when it found a difference.  Make that
a non-fatal initialization error.

Issue: GMIN-880
Change-Id: Iaf6565ddd400a1151582b6d107646719dafbd794
Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
index af5d769..a298792 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css2400b0_v21/sh_css_firmware.c
@@ -226,6 +226,9 @@ sh_css_load_firmware(const char *fw_data,
 #if (!defined HRT_CSIM && !defined HRT_RTL)
 		IA_CSS_ERROR("CSS code version (%s) and firmware version (%s) mismatch!",
 				file_header->version, release_version);
+		pr_err("ISP firmware version mismatch, got %s, want %s\n",
+		       file_header->version, release_version);
+		WARN_ON(1);
 		return IA_CSS_ERR_VERSION_MISMATCH;
 #endif
 	} else {
-- 
1.9.3

